service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/mapper/diagnosticreport/DiagnosticReportMapperTest.java

```diff
@@ -201,6 +206,52 @@ void When_DiagnosticReport_With_RedactedFilingComment_Expect_ConfidentialityCode
         assertThat(actualXml).containsIgnoringWhitespaces(NOPAT_HL7_CONFIDENTIALITY_CODE);
     }
 
+    @Test
+    void When_DiagnosticReport_With_ObservationEffectiveDateTimeTypeAndCommentNote_Expect_NarrativeStatementConfidentialityCodePresent() {
+        final String diagnosticReportFileName = "diagnostic-report-with-effective-datetime-filing-comment.json";
+        final DiagnosticReport diagnosticReport = getDiagnosticReportResourceFromJson(diagnosticReportFileName);
+        final Bundle bundle = getBundleResourceFromJson(INPUT_JSON_BUNDLE_WITH_FILING_COMMENTS);
+        final Observation observation = (Observation) bundle.getEntry().get(2).getResource();
+        final InputBundle inputBundle = new InputBundle(bundle);
+        final List<String> expectedXPaths = Collections.singletonList(
+            "/component/CompoundStatement/component[1]/NarrativeStatement/" + getNopatConfidentialityCodeXpathSegment()
+        );
+
+        when(confidentialityService.generateConfidentialityCode(observation))
```

Comment:
I think this should be OK as one-liners, takes less space and gives the same readability effect